tools/python/util/mobile_helpers/usability_checker.py

```diff
@@ -0,0 +1,558 @@
+                            self.nodes_unsupported_due_to_dynamic_input)
+
+        if logger.getEffectiveLevel() >= logging.DEBUG:
+            # Enable this manually if you need to look at specific partitions.
+            # for group in supported_groups:
+            #     logger.debug(f'Nodes in group: {",".join([f"{node.name}:{node.op_type}" for node in group])}')
+            if self.unsupported_ops:
+                logger.info(f'Unsupported ops: {",".join(sorted(self.unsupported_ops))}')
+
+            caveats = self.supported_ops_checker.get_caveats()
+            if caveats:
+                indent = ' ' * 5
+                logger.debug('Caveats that have not been checked and may result in a node not being supported:  '
+                             f'{"".join([os.linesep + indent + caveat for caveat in caveats])}')
+
+        pct_nodes_using_ep = self.num_supported_nodes / num_nodes * 100
+        if self.num_partitions == 0:
+            logger.info(f"{ep_name} can not run any nodes in this model.")
+        elif self.num_partitions == 1:
+            if pct_nodes_using_ep > 75:
+                logger.info(f"{ep_name} should work well for this model as there is one partition "
+                            f"covering {pct_nodes_using_ep}% of the nodes in the model.")
+            elif pct_nodes_using_ep > 50:
+                logger.info(
+                    f"{ep_name} may work well for this model, however only {pct_nodes_using_ep}% of nodes will use it. "
+                    "Performance testing is required to validate.")
+            else:
+                logger.info(
+                    f"{ep_name} will probably not work will for this model as only {pct_nodes_using_ep}% "
+                    "of nodes will use it.")
+
+        elif self.num_partitions == 2 and pct_nodes_using_ep > 75:
+            logger.info(f"{ep_name} can be considered for this model as there are two partitions "
+                        f"covering {pct_nodes_using_ep}% of the nodes. Performance testing is required to validate.")
+        else:
+            logger.info(f"{ep_name} is not recommended with this model as there are {self.num_partitions} partitions "
+                        f"covering {pct_nodes_using_ep}% of the nodes in the model. "
+                        "This will most likely result in worse performance than just using the CPU EP.")
+
+
+def check_partitioning(graph: onnx.GraphProto, supported_ops_checker: _SupportedOpsChecker,
+                       require_fixed_input_sizes: bool = False, value_info: dict = None):
+    '''
+    Estimate the partitions the graph will be split into for nodes that is_node_supported_fn returns true for.
+
+    The check on whether a node is supported is purely based on the operator type. Additional limitations
+    (e.g. NNAPI EP only supports 2D Conv) are not checked, so partitions may not be 100% accurate. The limitations
```

Comment:
Just a thought, can we consider some API in the future, can perform the actual partitioning and return the information? Or some API can retrieve information from an inference session
This can be useful for developers to decide if the partitioning is optimal at runtime

Reply:
Possible. But creating an InferenceSession with and without the EP enabled and running a few test inferences is probably going to be more reliable in terms of checking performance.